From 86b045e8d637a0296238f98d3d6568d226f6aff7 Mon Sep 17 00:00:00 2001
Message-Id: <86b045e8d637a0296238f98d3d6568d226f6aff7.1414143271.git.chang-joon.lee@intel.com>
In-Reply-To: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
References: <f393e797ccaaf66a25ee3d7c4e6136083a6effeb.1414143271.git.chang-joon.lee@intel.com>
From: Vidya Srinivas <vidya.srinivas@intel.com>
Date: Mon, 20 Oct 2014 20:26:04 +0530
Subject: [PATCH 25/26] FOR_UPSTREAM [VPG]: Fix in the IRQ handler for vblank
 interrupt

Current code doesnt give vblank events for tests like vbltest.
Patch fixes this issue by using START of vsync as event.

Issue: GMINL-3539
Change-Id: I14be3cdf8abcf2c8a1a19548149db4aafdcbbd9b
Signed-off-by: Vidya Srinivas <vidya.srinivas@intel.com>
---
 drivers/gpu/drm/i915/i915_irq.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/i915/i915_irq.c b/drivers/gpu/drm/i915/i915_irq.c
index bb642cb..580b28d 100644
--- a/drivers/gpu/drm/i915/i915_irq.c
+++ b/drivers/gpu/drm/i915/i915_irq.c
@@ -2096,7 +2096,7 @@ static void valleyview_pipestat_irq_handler(struct drm_device *dev, u32 iir)
 	spin_unlock(&dev_priv->irq_lock);
 
 	for_each_pipe(pipe) {
-		if (pipe_stats[pipe] & PIPE_VBLANK_INTERRUPT_STATUS) {
+		if (pipe_stats[pipe] & PIPE_START_VBLANK_INTERRUPT_STATUS) {
 			if (dev_priv->pf_change_status[pipe] &
 					BPP_CHANGED_PRIMARY)
 				I915_WRITE_BITS(VLV_DDL(pipe),
-- 
1.7.9.5

